pkg/subctl/lighthouse/deploy/ensure.go

```diff
@@ -17,9 +17,33 @@ limitations under the License.
 package lighthouse
 
 import (
+	"github.com/submariner-io/submariner-operator/pkg/internal/cli"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/lighthouse/install"
 	"k8s.io/client-go/rest"
 )
 
-func Ensure(config *rest.Config, repo string, version string) error {
-	return nil
+const (
+	DefaultControllerImageName    = "lighthouse-controller"
+	DefaultControllerImageRepo    = "quay.io/submariner"
+	DefaultControllerImageVersion = "0.0.1"
+)
+
+func Ensure(status *cli.Status, config *rest.Config, repo string, version string, isController bool) error {
+	image := ""
+	// TBD: Ensure federation
+	// Ensure lighthouse
+	if isController {
+		image = generateImageName(repo, DefaultControllerImageName, version)
+	}
+	return install.Ensure(status, config, image, isController)
+}
+
+func generateImageName(repo string, name string, version string) string {
```

Comment:
This is duplicating code we already have somewhere else. Can we refactor this to a common module?

Reply:
same as above, Stephen's patch addresses that.